inc/ticket.class.php

```diff
@@ -2565,6 +2579,73 @@ static function showMassiveActionsSubForm(MassiveAction $ma) {
             ]);
             echo "</td></tr></table>";
             return true;
+
+         case 'link_to_problem':
+            Problem::dropdown(['name' => 'problems_id']);
+            echo '<br><br>';
+            echo Html::submit(__('Link'), [
+               'name'      => 'link'
+            ]);
+            return true;
+
+         case 'resolve_tickets':
+            $rand_type = mt_rand();
+            echo '<div class="horizontal-form">';
+
+            echo '<div class="form-row">';
+            $label = _n('Solution template', 'Solution templates', 1);
+            echo "<label for='solution_template'>$label</label>";
+            SolutionTemplate::dropdown([
+               'name'     => "solution_template",
+               'value'    => 0,
+               // Load type and solution from bookmark
+               'toupdate' => [
+                  'value_fieldname' => 'value',
+                  'to_update'       => 'solution',
+                  'url'             => $CFG_GLPI["root_doc"]. "/ajax/solution.php",
+                  'moreparams' => [
+                     'type_id' => "dropdown_solutiontypes_id$rand_type"
+                  ]
+               ]
+            ]);
+            echo '</div>'; // .form-row
+
+            echo '<div class="form-row">';
+            $label = __('Solution type');
```

Comment:
Will cause gettext warnings.
```suggestion
            $label = SolutionType::getTypeName(1);
```